packages/evolution-frontend/tests/ui-testing/testHelpers.ts

```diff
@@ -975,3 +994,60 @@ export const pageRedirectionTest: RedirectionTest = ({ context, buttonText, expe
         expect(logoutCalled).toBe(true);
     });
 };
+
+/**
+ * Verify that a button in the navigation bar has the expected status by checking its class.
+ * Also checks if the button is disabled, and thus unclickable.
+ *
+ * @param {Object} options - The options for the test.
+ * @param {string} options.buttonText - The text of the nav bar button we are checking.
+ * @param {string} options.buttonStatus - The status we expect for the button. Can be one of four values: "completed", "active", "active and completed", and "inactive".
+ * @param {boolean} options.isDisabled - true if we expect the button to be disabled, false is not.
+ */
+export const verifyNavBarButtonStatus: NavBarButtonStatusTest = ({ context, buttonText, buttonStatus, isDisabled }) => {
+    test(`The ${buttonText} navigation button should have the '${buttonStatus}' status - ${getTestCounter(context, `${buttonText} - ${buttonStatus}`)}`, async () => {
+        const navBar = context.page.locator("div[class='survey-section-nav']");
+        const button = navBar.getByText(buttonText);
+        let expectedStatusClass;
+        switch(buttonStatus) {
+            case "completed":
+                expectedStatusClass = "nav-button completed-section";
+                break;
+            case "active":
+                expectedStatusClass = "nav-button active-section";
+                break;
+            case "activeAndCompleted":
+                expectedStatusClass = "nav-button active-section completed-section";
+                break;
+            case "inactive":
+                expectedStatusClass = "nav-button";
+                break;
+        }
+
+        expect(button).toHaveClass(expectedStatusClass);
```

Comment:
Est-ce que ce test va tester exclusivement ces status? par exemple, pour 'inactive', c'est juste `nav-button`, mais si le bouton a des classes en plus, genre 'active-section', est-ce que le test va échouer? Ou il faudrait ajouter des tests pour `not.toHaveClass`?

Reply:
Oui quand tu met un string ça fait une comparaison exacte. T'as aussi l'option de mettre du regex si tu veux un match partiel.